### PR TITLE
fix(astro): make the script element use typescript instead of javascript

### PR DESCRIPTION
**Pull request by virchau13**
The Astro parser should use the typescript parser for the `<script>` element instead of the javascript parser as defined by `; inherits: html`. This overrides it.

Note: Is this the correct way to do things? Does this completely override the HTML `javascript` injection or will this break at some point?

### CODE REVIEW

**Comment by virchau13** [comment]
Hey this is kind of important for Astro users to work properly with Typescript, is there anything blocking the merge of this PR? I'll be happy to submit code PRs if necessary

**Review by clason — APPROVED** [review]

**Comment by clason** [comment]
As you are the astro maintainer, I trust you on this. If it breaks, I'll ping you :)

**Pull request merged**
